APIs/APIManager.m

```diff
@@ -85,11 +85,23 @@ - (void)GMSPlaceFromID:(NSString*)placeID withCompletion:(void(^)(GMSPlace* plac
      ];
 }
 
--(void)getAllGMSPlaces:(void(^)(NSArray *places))completion {
+-(void)getAllGMSPlaces:(void(^)(NSMutableArray *places))completion {
     PFQuery *query = [PFQuery queryWithClassName:@"Place"];
     [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
         if (error == nil && objects != nil) {
-            completion(objects);
+            
+            // convert array of Place objects to GMSPlace objects
+            NSMutableArray *array = [[NSMutableArray alloc] init];
+            for (Place *myPlace in objects) {
+                [self GMSPlaceFromPlace:myPlace withCompletion:^(GMSPlace *place) {
+                    [array addObject:place];
+                    
+                    if(array.count == objects.count)
+                    {
+                        completion(array);
```

Comment:
Once we're done mutating our array (adding objects here), let's turn it into a NSArray. We prefer to pass immutable data if possible :)